### PR TITLE
Updating version number in Swagger -> SDK config.

### PR DESCRIPTION
**Pull request by marstr**
UserAgent was also updated.

---

Thanks you for your contribution to the Azure-SDK-for-Go! We will triage and review it as quickly as we can.

As part of your submission, please make sure that you can make the following assertions:

 - [x] I'm not making changes to Auto-Generated files which will just get erased next time there's a release.
   - If that's what you want to do, consider making a contribution here: https://github.com/Azure/autorest.go
 - [ ] I've tested my changes, adding unit tests where applicable.
 - [ ] I've added Apache 2.0 Headers to the top of any new source files.
 - [x] I'm submitting this PR to the `dev` branch, or I'm fixing a bug that warrants its own release and I'm targeting the `master` branch.
 - [ ] If I'm targeting the `master` branch, I've also added a note to [CHANGELOG.md](https://github.com/Azure/azure-sdk-for-go/blob/master/README.md).
 - [ ] I've mentioned any relevant open issues in this PR, making clear the context for the contribution.

### CODE REVIEW

File: swagger_to_sdk_config.json [hunk 1]
```diff
@@ -9,8 +9,8 @@
       "verbose": "",
       "sdkrel:go-sdk-folder": ".",
       "multiapi": "",
-      "package-version": "v12.2.1-beta",
-      "user-agent": "Azure-SDK-For-Go/v12.2.1-beta services"
+      "package-version": "v12.3.1-beta",
```
Comment (jhendrixMSFT) [inline swagger_to_sdk_config.json]: Why is this v12.3.1-beta?  The latest release is v12.3.0-beta.

Comment (marstr) [reply swagger_to_sdk_config.json]: Righto, I deliberately have it incremented by one so that it's potentially ready to go if we need to release.

Comment (jhendrixMSFT) [reply swagger_to_sdk_config.json]: I see.  What is this file used for?

Comment (marstr) [reply swagger_to_sdk_config.json]: This is a configuration file for the Swagger -> SDK bot.

Comment (jhendrixMSFT) [reply swagger_to_sdk_config.json]: This is the per commit thing or something else?  When does a person have to update this version info?

Comment (vladbarosan) [reply swagger_to_sdk_config.json]: @marstr per @jhendrixMSFT  comments I think this raises the issues to properly document in a readme or in OneNote how to use the configs and the generator.

Comment (marstr) [reply swagger_to_sdk_config.json]: The idea was that temporarily, this file would stay one ahead of the released version in anticipation of the changes being released as patches. Ideally the version number would be automatically set, but we had trouble doing it that way.

I was going to continue updating it, but it looks like I may have a little spare time this morning to skip this step and instead submit a PR with all of the configuration changes we were looking for.

Comment (jhendrixMSFT) [reply swagger_to_sdk_config.json]: I thought we were going to use a different version format, something like 20180201.00-nightly or something.  Did that change?

Comment (marstr) [reply swagger_to_sdk_config.json]: I'm going to cross-post some of this conversation to the new PR that I opened up.

**Pull request closed without merging**